src/devices/Card/CreditCard/CreditCard.cs

```diff
@@ -0,0 +1,697 @@
+                                        TerminalTransactionQualifier.ConsumerDeviceCvmSupported | TerminalTransactionQualifier.IssuerUpdateProcessingSupported;
+                                    // Encode the TTq
+                                    BinaryPrimitives.TryWriteUInt32BigEndian(toSend.Slice(index, 4), (uint)ttq);
+                                }
+                                // Transaction amount
+                                else if (dol.TagNumber == 0x9F02)
+                                {
+                                    // Ask authorization for the minimum, just to make sure
+                                    // It's more than 0
+                                    toSend[index + 5] = 1;
+                                }
+                                // 9F1A-Terminal Country Code,
+                                else if (dol.TagNumber == 0x9F1A)
+                                {
+                                    // Let's say we are in France
+                                    toSend[index] = 0x02;
+                                    toSend[index + 1] = 0x50;
+                                }
+                                // 009A-Transaction Date
+                                else if (dol.TagNumber == 0x9A)
+                                {
+                                    toSend[index] = NumberHelper.Dec2Bcd((DateTime.Now.Year % 100));
+                                    toSend[index + 1] = NumberHelper.Dec2Bcd((DateTime.Now.Month));
+                                    toSend[index + 2] = NumberHelper.Dec2Bcd((DateTime.Now.Day));
+                                }
+                                // 0x9F37 Unpredictable number
+                                else if (dol.TagNumber == 0x9F37)
+                                {
+                                    var rand = new Random();
+                                    rand.NextBytes(toSend.Slice(index, dol.Data[0]));
+                                }
+                                // Currency
+                                else if (dol.TagNumber == 0x5F2A)
+                                {
+                                    // We will ask for Euro
+                                    toSend[index] = 0x09;
+                                    toSend[index + 1] = 0x78;
+                                }
+
+                                index += dol.Data[0];
+                            }
+                        }
+
+                        // Ask for all the process options
+                        ret = GetProcessingOptions(toSend, received);
+                        Tag appLocator = null;
+                        if (ret == ErrorType.ProcessCompletedNormal)
+                        {
+                            // Check if we have an Application File Locator 0x94 in 0x77
```

Comment:
not sure I follow what the comment means, could you add some spec page here?

Reply:
Explanation in EMV 4.3 Book 3, Integrated Circuit Card Specifications for Payment Systems. https://www.emvco.com/emv-technologies/contact/. Not really a specific page, it's about reading a lot to understand how things are related, mainly chapter 5 and chapter 7. I will add in the description then.